letsencrypt/dhparam.sls

```diff
@@ -0,0 +1,28 @@
+#!jinja|yaml
+# -*- coding: utf-8 -*-
+# vim: ft=sls
+
+{% from "letsencrypt/map.jinja" import letsencrypt with context %}
+
+{%- if letsencrypt.dhparam.enable -%}
+
+{%- if letsencrypt.dhparam.global_dhparam -%}
+letsencrypt_dhparam_global:
+  cmd.run:
+    - name: /usr/bin/openssl dhparam -out /etc/letsencrypt/live/dhparam.pem {{ letsencrypt.dhparam.keylength }}
+    - creates: /etc/letsencrypt/live/dhparam.pem
+{%- else -%}
+{%- endif -%}
```

Comment:
This `endif` will result in a syntax error for this state because it is closing the inner `if` block before the `else` can be executed. It should be moved down to line 27.